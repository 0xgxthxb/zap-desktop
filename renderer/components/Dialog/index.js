export DialogAppCrashed from './DialogAppCrashed'
export DialogLndCrashed from './DialogLndCrashed'
