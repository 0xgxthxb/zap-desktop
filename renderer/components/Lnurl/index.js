export LnurlAuthPrompt from './LnurlAuthPrompt'
export LnurlChannelPrompt from './LnurlChannelPrompt'
