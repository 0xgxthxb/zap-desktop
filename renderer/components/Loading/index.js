import LoadingBolt from './Loading'

export default LoadingBolt
