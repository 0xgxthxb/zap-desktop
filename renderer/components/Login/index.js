import Login from './Login'

export LoginNotAllowed from './LoginNotAllowed'
export default Login
