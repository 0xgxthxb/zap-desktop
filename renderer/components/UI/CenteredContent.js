import React from 'react'
import { Flex } from 'rebass'

const CenteredContent = props => (
  <Flex
    alignItems="center"
    flexDirection="column"
    height="100%"
    justifyContent="center"
    width={1}
    {...props}
  />
)

export default CenteredContent
