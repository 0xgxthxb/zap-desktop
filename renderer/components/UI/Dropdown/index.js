import Dropdown from './Dropdown'

export { DropdownButton } from './Dropdown'
export default Dropdown
