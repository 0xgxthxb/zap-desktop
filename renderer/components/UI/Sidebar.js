import React from 'react'
import PropTypes from 'prop-types'
import { Flex } from 'rebass/styled-components'

const Sidebar = ({ hasShadow, ...rest }) => (
  <Flex
    as="aside"
    color="primaryText"
    flexDirection="column"
    width={4 / 12}
    {...rest}
    sx={{
      overflow: 'hidden',
      boxShadow: hasShadow ? 'm' : null,
      zIndex: 1,
    }}
  />
)

Sidebar.propTypes = {
  hasShadow: PropTypes.bool,
}

Sidebar.defaultProps = {
  hasShadow: true,
}

Sidebar.small = props => <Sidebar {...props} width={3 / 16} />
Sidebar.medium = props => <Sidebar {...props} width={4 / 16} />
Sidebar.large = props => <Sidebar {...props} width={6 / 16} />

Sidebar.small.displayName = 'Sidebar Small'
Sidebar.medium.displayName = 'Sidebar Medium'
Sidebar.large.displayName = 'Sidebar Large'

export default Sidebar
