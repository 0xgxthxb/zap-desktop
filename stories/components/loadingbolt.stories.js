import React from 'react'
import { storiesOf } from '@storybook/react'
import { boolean } from '@storybook/addon-knobs'
import { Page } from 'components/UI'
import Loading from 'components/LoadingBolt'

storiesOf('Components', module).addWithChapters('LoadingBolt', {
  subtitle: 'Animation to indicate application is loading.',
  chapters: [
    {
      sections: [
        {
          title: 'Bolt',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading hasClose isLoading={isLoading} onClose={() => {}} variant="bolt" />
              </Page>
            )
          },
        },
        {
          title: 'App',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading hasClose isLoading={isLoading} onClose={() => {}} variant="app" />
              </Page>
            )
          },
        },
        {
          title: 'Launchpad',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading isLoading={isLoading} variant="launchpad" />
              </Page>
            )
          },
        },
      ],
    },
  ],
})
