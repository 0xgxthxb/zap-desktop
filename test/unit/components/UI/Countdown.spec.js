import React from 'react'
import { IntlProvider } from 'react-intl'
import { renderWithTheme } from '@zap/test/unit/__helpers__/renderWithTheme'
import { Countdown } from 'components/UI'

describe('component.UI.Countdown', () => {
  it('should render correctly with expiry set to a date in the past', () => {
    const tree = renderWithTheme(
      <IntlProvider locale="en">
        <Countdown offset={new Date('2009-01-03T18:15:05+00:00')} />
      </IntlProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with expiry in 60 seconds', () => {
    const tree = renderWithTheme(
      <IntlProvider locale="en">
        <Countdown offset={60} />
      </IntlProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with expiry in 365 days', () => {
    const tree = renderWithTheme(
      <IntlProvider locale="en">
        <Countdown offset={86400 * 365} />
      </IntlProvider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
