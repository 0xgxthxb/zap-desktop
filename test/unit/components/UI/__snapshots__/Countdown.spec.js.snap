// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Countdown should render correctly with expiry in 60 seconds 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #39e673;
  font-size: 13px;
  line-height: 1.4;
}

<div
  className="c0"
  color="superGreen"
  fontSize="m"
>
  Expires
  <i>
     
  </i>
  in 1 minute
</div>
`;

exports[`component.UI.Countdown should render correctly with expiry in 365 days 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #39e673;
  font-size: 13px;
  line-height: 1.4;
}

<div
  className="c0"
  color="superGreen"
  fontSize="m"
>
  Expires
  <i>
     
  </i>
  in 365 days
</div>
`;

exports[`component.UI.Countdown should render correctly with expiry set to a date 1 hour in the past 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: #e63939;
  font-size: 13px;
  line-height: 1.4;
}

<div
  className="c0"
  color="superRed"
  fontSize="m"
>
  Expired
  <i>
     
  </i>
  1 hour ago
</div>
`;
